OpenRA.Game/Graphics/SpriteRenderer.cs

```diff
@@ -108,6 +109,21 @@ public void DrawSprite(Sprite s, float2 a, float2 b, float2 c, float2 d)
 			nv += 4;
 		}
 
+		public void DrawSprite(Sprite s, Vertex[] sourceVertices, int offset)
+		{
+			renderer.CurrentBatchRenderer = this;
+
+			if (s.Sheet != currentSheet ||
+				s.BlendMode != currentBlend ||
+				nv + 4 > renderer.TempBufferSize)
+				Flush();
+
+			currentBlend = s.BlendMode;
+			currentSheet = s.Sheet;
+			Array.Copy(sourceVertices, offset, vertices, nv, 4);
```

Comment:
This is essentially inlining the equivalent to the `Util.FastCreateQuad` above.  It would be nice if we could merge or otherwise improve the consistency.